### PR TITLE
[Impeller] Move drawAtlas golden tests to display list.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/142054

Moves all of the drawAtlas tests for aiks and entities into DL. This might change some of the entity goldens but ... these are easy to eyeball so it should be fine IMO.

### CODE REVIEW

File: impeller/display_list/aiks_dl_atlas_unittests.cc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+}  // namespace
+
+TEST_P(AiksTest, DrawAtlasNoColor) {
+  DisplayListBuilder builder;
+  auto [texture_coordinates, transforms, atlas] = CreateTestData(this);
+
+  builder.Scale(GetContentScale().x, GetContentScale().y);
+  builder.DrawAtlas(atlas, transforms.data(), texture_coordinates.data(),
+                    nullptr, 4, DlBlendMode::kSrcOver,
```
Comment (gaaclarke) [inline impeller/display_list/aiks_dl_atlas_unittests.cc:58]:
```suggestion
                    /*colors=*/nullptr, /*count=*/4, DlBlendMode::kSrcOver,
```

File: impeller/display_list/aiks_dl_atlas_unittests.cc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                    DlImageSampling::kNearestNeighbor, nullptr);
```
Comment (gaaclarke) [inline impeller/display_list/aiks_dl_atlas_unittests.cc:59]:
```suggestion
                    DlImageSampling::kNearestNeighbor, /*cullRect=*/nullptr);
```

File: impeller/display_list/aiks_dl_atlas_unittests.cc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, DrawAtlasNoColorFullSize) {
+  auto atlas = DlImageImpeller::Make(CreateTextureForFixture("bay_bridge.jpg"));
+  auto size = atlas->impeller_texture()->GetSize();
+  std::vector<SkRect> texture_coordinates = {
+      SkRect::MakeLTRB(0, 0, size.width, size.height)};
+  std::vector<SkRSXform> transforms = {MakeTranslation(0, 0)};
```
Comment (gaaclarke) [inline impeller/display_list/aiks_dl_atlas_unittests.cc:116]: No action required:  "SkRSXform", really skia?

**Review by gaaclarke — APPROVED** [review]
LGTM, I think parameter comments would help reading the tests but they weren't there before so no requirement to add them.

**Comment by gaaclarke** [comment]
Goldens look reasonable to me too, nothing was obviously broken to me.

File: impeller/display_list/aiks_dl_atlas_unittests.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_atlas_unittests.cc]: Done

File: impeller/display_list/aiks_dl_atlas_unittests.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_atlas_unittests.cc:59]: Done

**Pull request merged**